### DISCUSSION TITLE
Use case or model for Load, Battery and Solar PV

### DISCUSSION THREAD

**Discussion by TheASingh** [CURENT/andes#264] [Modeling] [answered]
Good morning Dr. Cui,

I have a requirement to use **Loads**, **Batteries** and **Solar PVs** in the **simulation**.

Please suggest, which **use-case / model**  is available to achieve simulation for above mentioned devices?

Regards,
A Singh

**Comment by cuihantao** [comment, marked as answer]
Hi @TheASingh 

You can go over the model library at https://docs.andes.app/en/stable/modelref.html. The ones starting with `Ren` are for renewables. Load models are in https://docs.andes.app/en/stable/groupdoc/DynLoad.html. 

To model solar PVs, you can follow the example in `cases/ieee14/ieee14_solar.xlsx`. We don't have a battery model yet.

Please excuse me for not having the capacity to help with any case creation. You are encouraged to read the source code.

**Reply by TheASingh** [reply]
Hi Dr. Cui,

As per our use case, we need to include **batteries** as well. 

Has someone developed/created a model for **battery** so far in ANDES?

Or is it possible to develop a new model for **battery**? If yes, please let me know about the further steps?

Regards,
A Singh

**Reply by cuihantao** [reply]
No specific battery model yet. There is an `ESD1` model for distributed energy storage: https://docs.andes.app/en/stable/groupdoc/DG.html#esd1. This is just generic storage and no specifics for battery.

You can develop new models by going through the Modeling Cookbook and existing code examples.

**Reply by amparocorvalan** [reply]
Hi @TheASingh, 

Did you develop a battery model? I also need it for a system I want to study.

Regards,
Amparo

**Reply by amparocorvalan** [reply]
Hi @cuihantao, 

Where can I find the Modeling Cookbook?

Regards,
Amparo

**Reply by cuihantao** [reply]
It's renamed to the Development section here: https://docs.andes.app/en/latest/modeling/index.html

**Reply by amparocorvalan** [reply]
Thanks!
